kgforge/core/archetypes/store.py

```diff
@@ -378,44 +378,49 @@ def search(
         #   - lookup: str, with values in ('current', 'children').
         # POLICY Should use sparql() when 'sparql' is chosen as value  for the param 'search_endpoint'.
         # POLICY Should use elastic() when 'elastic' is chosen as value  for the param 'search_endpoint'.
+        # POLICY Given parameters for limit and offset override the input query.
         # POLICY Should notify of failures with exception QueryingError including a message.
         # POLICY Resource _store_metadata should be set using wrappers.dict.wrap_dict().
         # POLICY Resource _synchronized should be set to True.
         # TODO These two operations might be abstracted here when other stores will be implemented.
         not_supported()
 
     def sparql(
-        self, query: str, debug: bool, limit: int = None, offset: int = None, **params
+        self, query: str, debug: bool, limit: int = 100, offset: int = 0, **params
     ) -> List[Resource]:
         rewrite = params.get("rewrite", True)
         qr = (
             rewrite_sparql(query, self.model_context, self.service.metadata_context)
             if self.model_context is not None and rewrite
             else query
         )
-        qr = _replace_in_sparql(qr, "LIMIT", limit, 100, r" LIMIT \d+")
-        qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
+        if limit:
+            qr = _replace_in_sparql(qr, "OFFSET", offset, 0, r" OFFSET \d+")
```

Comment:
You swapped the conditions and the corresponding replacing statements

Reply:
Also this is more of a format thing than a functionality thing, but 100 and 0 are hardcoded in many locations in these files, maybe having a global variable like SPARQL_DEFAULT_LIMIT & ELASTIC_DEFAULT_LIMIT + same for offset (or just DEFAULT_LIMIT and DEFAULT_OFFSET if it's known that both will always have the same defaults) could be better?

Reply:
yeah, DEFAULT_LIMIT and DEFAULT_OFFSET could work 👍🏽

Reply:
If a limit is hardcoded, but a limit is specified in the function, the function value should override the hardcoded limit. This works when a number is specified, but if it is specified that the limit should be None, it keeps the hardcoded value. This feels unintentional. I'd think the limit should be removed altogether, no?